docs/python.md

```diff
@@ -103,6 +103,19 @@ The `Input()` function takes these keyword arguments:
 
 Each parameter of the `predict()` method must be annotated with a type like `str`, `int`, `float`, `bool`, etc. See [Input and output types](#input-and-output-types) for the full list of supported types.
 
+Using the `Input` function is recommended because it provides better documentation and validation constraints to the users of your model, but it is not strictly required. You can also specify default values for your parameters using plain Python, or omit default assignment entirely:
```

Comment:
Not a big deal, but I'm not sure it is recommended? You get validation just with plain Python because of the type annotations. I really like the fact that it's Just Python Type Annotations, because it makes it understandable. Then, you can layer on the more complicated `cog.Input()` thing on top of that if you need extra annotation.

This really is a very minor point though so maybe over time we can iterate on this.

Reply:
E.g. look at the headline examples on pydantic and fastapi:

https://pydantic-docs.helpmanual.io/
https://fastapi.tiangolo.com/

They don't use any of their clever default functions anywhere, and it makes it look more like simple, understandable, vanilla Python.

Reply:
I think it's recommended primarily because we want the descriptions to propagate through to the OpenAPI schema, to API clients, to CLIs, to docs, for the benefit of end users.

Reply:
Changed the wording a bit here and removed the "recommended" bit.

Reply:
Yeah, I suppose encouraging people to write descriptions is a good reason to encourage `Input`.